tests/suites/test_suite_pk.function

```diff
@@ -493,9 +496,12 @@ void pk_rsa_alt( )
     TEST_ASSERT( mbedtls_pk_sign( &alt, MBEDTLS_MD_NONE, hash, sizeof hash,
                           sig, &sig_len, rnd_std_rand, NULL ) == 0 );
 #if defined(MBEDTLS_HAVE_INT64)
-    TEST_ASSERT( mbedtls_pk_sign( &alt, MBEDTLS_MD_NONE, hash, (size_t)-1,
-                          NULL, NULL, rnd_std_rand, NULL ) ==
-                 MBEDTLS_ERR_PK_BAD_INPUT_DATA );
+    if( sizeof( size_t ) > sizeof( unsigned int ) )
+    {
+        TEST_ASSERT( mbedtls_pk_sign( &alt, MBEDTLS_MD_NONE, hash, (size_t)-1,
+                              NULL, NULL, rnd_std_rand, NULL ) ==
```

Comment:
This (preexisting) test is weird: it passes null pointers as the `sig` and `sig_len` arguments to `mbedtls_pk_sign`, which is not allowed. It happens that `mbedtls_pk_sign` validates `hash_len` before `sig` and `sig_len`, but relying on this makes the test case fragile. The test case should be changed so that `hash_len` is the only invalid parameter.